ingestion_server/ingestion_server/api.py

```diff
@@ -129,39 +131,43 @@ class TaskStatus(BaseTaskResource):
     def on_get(self, req, resp, task_id):
         """Check the status of a single task."""
         task = self.tracker.id_task[task_id]
-        active = task.is_alive()
 
         percent_completed = self.tracker.id_progress[task_id].value
+        active_workers = bool(self.tracker.id_active_workers[task_id].value)
+        active = task.is_alive() or active_workers
+
         resp.media = {
             "active": active,
             "percent_completed": percent_completed,
             "error": percent_completed < 100 and not active,
```

Comment:
These changes make `error` slightly more accurate than it was before, but it's still not great:

* `error` will now correctly be `True` if `percent_completed < 100` and the _task died before starting the reindexing workers_
* however, we're currently only setting `active_workers` to False when the `go_live` step completes successfully. 

The indexer workers themselves wrap their work in a try/except and report back to the ingestion server [even if they encounter errors](https://github.com/WordPress/openverse-api/blob/main/ingestion_server/ingestion_server/indexer_worker.py#L84). I am fairly certain this means that `go_live` will run even if the re-indexing step fails, and that we expect to catch this at [this healthcheck step](https://github.com/WordPress/openverse-api/blob/main/ingestion_server/ingestion_server/indexer.py#L359) within the method. I _think_ what we'll want to do is update `active_workers` to false if that healthcheck times out, but I may be misreading how this works.

Reply:
After looking into it further, I don't understand the current state of our error handling and suggest that it might be best to leave updating that for another ticket -- or to some extent, leaving it to be handled sensibly when we move the logic to Airflow. With this work `error` here should be at least improved.

* As I already stated, I believe the indexer workers simply catch and log any errors they encounter, but when they complete the `go_live` step is called regardless
* `go_live` has a [healthcheck](https://github.com/WordPress/openverse-api/blob/main/ingestion_server/ingestion_server/indexer.py#L359) but it only runs in production, not staging. It waits for the es cluster to be GREEN, and will block for up to a 12 hour timeout. But as far as I can tell that check just returns when the timeout is reached and we aren't doing any sort of response checking -- so if it times out with the cluster still unhealthy, I'm pretty sure we just go on to replace the indices regardless 😰 

I can make a ticket/PR for that last point.

Reply:
Per my comment [here](https://github.com/WordPress/openverse-api/pull/593#issuecomment-1082287575), I've updated this so that workers report back to the ingestion server when they encounter errors, and `go_live` is not called unless all workers complete successfully.